service/src/main/java/bio/terra/externalcreds/services/LinkedAccountService.java

```diff
@@ -37,6 +39,13 @@ public LinkedAccountService(
     return linkedAccountDAO.getLinkedAccount(userId, providerId);
   }
 
+  @ReadTransaction
+  public Optional<GA4GHPassport> getGA4GHPassport(String userId, String providerId) {
+    var linkedAccount = linkedAccountDAO.getLinkedAccount(userId, providerId);
+    if (linkedAccount.isEmpty()) throw new NotFoundException("Linked account not found");
+    return ga4ghPassportDAO.getPassport(linkedAccount.get().getId().orElseThrow());
```

Comment:
This can be done in a single database query, not 2. It looks like `ga4ghPassportDAO.getPassport(linkedAccountId)` is only called by tests. Perhaps that function should be changed to accept userId and providerId as parameters instead.

Reply:
I think I got it this time…